Public/Src/Cache/ContentStore/Distributed/NuCache/EventStreaming/EventHubContentLocationEventStore.cs

```diff
@@ -265,6 +268,28 @@ async Task<BoolResult> sendToActionBlockAsync()
 
                 return BoolResult.Success;
             }
+
+            string extraEndMessageForSendToActionBlock()
+            {
+                int messageCount = 0;
+                int fullQueueCount = 0;
+                int emptyQueueCount = 0;
+                foreach (var ab in _eventProcessingBlocks)
+                {
+                    var count = ab.InputCount;
+                    messageCount += count;
+                    if (count == _configuration.EventProcessingMaxQueueSize)
+                    {
+                        fullQueueCount++;
+                    }
+                    else if (count == 0)
+                    {
+                        emptyQueueCount++;
+                    }
+                }
+
+                return $"OverallQueueSize={messageCount}, FullQueues={fullQueueCount}, EmptyQueues={emptyQueueCount}";

```

Comment:
>" [](start = 24, length = 1)

Add queue index for the current operation, max queue size (w/ index of queue) and min queue size (with index of queue)

Reply:
This is done once per the entire invocation of ProcessEventsAsync, i.e. after all the items were added to possibly multiple action blocks.
We may trace each and individual operation as well, but I think it should be a different tracing message (because the current function just augments the existing tracing message with very high level information).

I was thinking that this should be enough to understand the state of the queue (is it poorly balanced or not) without adding too much of extra tracing.

But I'm fine adding extra tracing per each call to actionBlock.SendAsync().

---
In reply to: [319760951](https://github.com/microsoft/BuildXL/pull/821#discussion_r319760951) [](ancestors = 319760951)